giraph-block-app-8/src/test/java/org/apache/giraph/block_app/framework/no_vtx/MessagesWithoutVerticesTest.java

```diff
@@ -0,0 +1,157 @@
+package org.apache.giraph.block_app.framework.no_vtx;
+
+import java.util.Iterator;
+
+import org.apache.giraph.block_app.framework.AbstractBlockFactory;
+import org.apache.giraph.block_app.framework.BlockUtils;
+import org.apache.giraph.block_app.framework.api.BlockWorkerReceiveApi;
+import org.apache.giraph.block_app.framework.api.BlockWorkerSendApi;
+import org.apache.giraph.block_app.framework.block.Block;
+import org.apache.giraph.block_app.framework.block.SequenceBlock;
+import org.apache.giraph.block_app.framework.piece.Piece;
+import org.apache.giraph.block_app.framework.piece.interfaces.VertexReceiver;
+import org.apache.giraph.block_app.framework.piece.interfaces.VertexSender;
+import org.apache.giraph.block_app.test_setup.TestGraphUtils;
+import org.apache.giraph.block_app.test_setup.graphs.EachVertexInit;
+import org.apache.giraph.block_app.test_setup.graphs.Small1GraphInit;
+import org.apache.giraph.conf.GiraphConfiguration;
+import org.apache.giraph.function.ObjectTransfer;
+import org.apache.giraph.graph.Vertex;
+import org.apache.giraph.types.ops.collections.array.WIntArrayList;
+import org.apache.hadoop.io.IntWritable;
+import org.apache.hadoop.io.NullWritable;
+import org.junit.Assert;
+import org.junit.Test;
+
+import com.google.common.collect.Iterables;
+
+public class MessagesWithoutVerticesTest {
+
+  @Test
+  public void testWithLocalBlockRunner() throws Exception {
+    testSumOverSameGroup(3, false);
+  }
+
+  @Test
+  public void testWithGiraphEnv() throws Exception {
+    testSumOverSameGroup(3, true);
+  }
+
+  private void testSumOverSameGroup(int max, boolean fullGiraphEnv) throws Exception {
+    TestGraphUtils.runTest(
+        TestGraphUtils.chainModifiers(
+            new Small1GraphInit<IntWritable, IntWritable, NullWritable>(),
+            new EachVertexInit<>((vertex) -> vertex.getValue().set(- vertex.getId().get() % max))),
```

Comment:
yes - and 0 is such vertex in the test, though test might not fail in that case - will add.